addons/mass_mailing/static/src/js/mass_mailing_mobile_preview.js

```diff
@@ -1,53 +1,46 @@
 /** @odoo-module **/
 
+import { Component, useEffect, useRef } from "@odoo/owl";
 import { Dialog } from "@web/core/dialog/dialog";
-import { getBundle } from "@web/core/assets";
+import { _t } from "@web/core/l10n/translation";
 
-const { useEffect, onWillStart } = owl;
+export class MassMailingMobilePreviewDialog extends Component {
+    static components = {
+        Dialog,
+    };
+    static template = "mass_mailing.MobilePreviewDialog";
+    static props = {
+        title: { type: String },
+        preview: { type: String },
+        close: Function,
+    };
 
+    appendPreview() {
+        const iframe = this.iframeRef.el.contentDocument;
+        const body = iframe.querySelector("body");
+        body.innerHTML = this.props.preview;
+    }
 
-export class MassMailingMobilePreviewDialog extends Dialog {
-    setup() {
-        super.setup();
-        onWillStart(async () => {
-            this.bundle = await getBundle("mass_mailing.iframe_css_assets_edit");
-        });
-        useEffect((modalEl) => {
-            if (modalEl) {
-                const modalBody = modalEl.querySelector('.modal-body');
-                const invertIcon = document.createElement("span");
-                invertIcon.className = "fa fa-refresh";
-                const iframe = document.createElement("iframe");
-                iframe.srcdoc = this._getSourceDocument();
+    get title() {
+        return _t("Mobile Preview");
```

Comment:
title props not used?

Reply:
Yes it is ... 
  ` <span class="modal-title text-break" t-esc="title"></span>`